pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -26,6 +28,33 @@
 s = Settings()
 
 
+def get_mean_positions(positions, cell, pbc, labels=None):
+    """
+    This function calculates the average position(-s) across periodic boundary conditions. If the
+    labels are specified, the positions are grouped accordingly.
+
+    Args:
+        positions (numpy.ndarray (n, 3)): Coordinates to be averaged
+        cell (numpy.ndarray (3, 3)): Cell dimensions
+        pbc (numpy.ndarray (3,)): Periodic boundary conditions (in boolean)
+        labels (numpy.ndarray (n,)): (optional) labels according to which the atoms are
+            grouped
+
+    Returns:
+        (numpy.ndarray): mean positions
+    """
+    if labels is None:
+        labels = np.zeros(len(positions)).astype(int)
+    _, labels, counts = np.unique(labels, return_inverse=True, return_counts=True)
+    mean_positions = positions[np.unique(labels, return_index=True)[1]]
+    all_positions = positions-mean_positions[labels]
+    all_positions = np.einsum('ji,nj->ni', np.linalg.inv(cell), all_positions)
+    all_positions -= np.rint(all_positions)[:,pbc]
```

Comment:
I added strings explaining what these lines do. I hope now it's easy to understand.

Reply:
I see your point. Indeed if the point cloud is larger than half the simulation box, it might break down. I share the problem in the figure below. In particular, the average position might depend on the reference point (from which all arrows are drawn). I'm not really sure if there's a good way to prevent this from happening from the beginning, except for calculating all possible average positions by taking all possible reference points. And I'm not really sure whether we should raise an error if the largest distance is close to half a box, because I don't know how to define "close".

Other than that, it `rint` should be a solid method whether certain positions are far outside the box or not, as far as I can imagine at least.

<img width="623" alt="Screenshot 2021-06-08 at 10 25 16" src="https://user-images.githubusercontent.com/37879103/121150481-e0835700-c843-11eb-8286-89b8ecd7a104.png">

Reply:
That is a super good sketch of the issue.

Is this not a case where you want the MIC atom relative to your reference point? In this case ASE's `find_mic` might do the trick more robustly instead of what you have here. I might still misunderstand the problem though. Geometry is always surprisingly difficult 😭

Reply:
Yeah ASE will have essentially the same problem. Depending on what you consider as the reference point, certain positions could be lying across different boundaries. So in the example above, I took a reference point of a position on the right hand side and calculate the average position, but if I take one of the positions on the left side (close to the border) as the reference point, I'll get a different average position.

Reply:
> So in the example above, I took a reference point of a position on the right hand side and calculate the average position, but if I take one of the positions on the left side (close to the border) as the reference point, I'll get a different average position.

Yes, but not if you wrap the result back into the home cell. Using `find_mic` with the reference point as the atom shown, I get the three atoms in at the right side of the cell, and three ghost atoms outside it -- breaking that long arrow to the left. Alternatively, suppose I choose that very atom at the head of the long arrow as my reference site. Then applying `find_mic` to the other atoms I will get the three atoms on the left side of the cell, plus three more ghost atoms to the left outside it. The centres of these two clusters will differ by Lx. I think that final cell length offset is unavoidable and maybe you want to wrap back into the home cell for the final result.

Reply:
Okay I have to admit that the figure above was quite useless XD. Here's my second try: Imagine a cluster of spheres below. If I take the green sphere as the reference point, I'd take the PBC into account (I put opacity to represent PBC) and the cluster center will be more or less where the green atom is, right? But if I take the red sphere as the reference point, the displacement vectors will be different, and the cluster center will probably be more or less where the red sphere is.

But it's true that regardless of this discussion indeed I should anyway use `find_mic` because the lines above do exactly the same thing.

<img width="417" alt="Screenshot 2021-06-08 at 16 29 23" src="https://user-images.githubusercontent.com/37879103/121203847-ba2adf00-c876-11eb-82eb-a85fa1d48547.png">

Reply:
Ok, yeah, I see your point now. I guess what you really need is a loop over available image sites and finally choose the first moment that comes from a set of sites that minimize the second moment?

Reply:
Yes, but I guess for now I'm gonna keep the current version because if I have to minimise the second moment it's going to be computationally expensive.